grpc-bridge-runtime/src/main/java/org/jboss/resteasy/grpc/runtime/servlet/AsyncContextImpl.java

```diff
@@ -0,0 +1,102 @@
+package org.jboss.resteasy.grpc.runtime.servlet;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+
+import org.jboss.resteasy.concurrent.ContextualExecutors;
+
+import jakarta.servlet.AsyncContext;
+import jakarta.servlet.AsyncListener;
+import jakarta.servlet.ServletContext;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.ServletRequest;
+import jakarta.servlet.ServletResponse;
+
+public class AsyncContextImpl implements AsyncContext {
+
+   private static ExecutorService executorService = ContextualExecutors.threadPool();
+
+   private ServletRequest servletRequest;
+   private ServletResponse servletResponse;
+   private Set<AsyncListener> listeners = new HashSet<AsyncListener>();
+   long timeout;
```

Comment:
Should this be private? It should probably at least be volatile.

Reply:
Done